pkg/generate/app/pipeline.go

```diff
@@ -237,6 +237,15 @@ func (p *Pipeline) Objects(accept, objectAccept Acceptor) (Objects, error) {
 		if objectAccept.Accept(build) {
 			objects = append(objects, build)
 		}
+		if p.Build.Source != nil && p.Build.Source.SourceImage != nil && accept.Accept(p.Build.Source.SourceImage) {
+			srcImage, err := p.Build.Source.SourceImage.ImageStream()
```

Comment:
Does this have to be an image stream?  I don't think so - I think this should require p.Build.Source.SourceImage.AsImageStream